File: problems/2190-most-frequent-number-following-key-in-an-array/solution.cpp
class Solution {
public:
    int mostFrequent(vector<int>& nums, int key) {
        unordered_map<int, int> count;
        for (int i = 0; i < nums.size() - 1; ++i) {
            if (nums[i] == key) {
                count[nums[i + 1]]++;
            }
        }
        int maxCount = 0;
        int result = -1;
        for (auto const& [key, val] : count) {
            if (val > maxCount) {
                maxCount = val;
                result = key;
            }
        }
        return result;
    }
};

File: problems/2190-most-frequent-number-following-key-in-an-array/solution.py
class Solution:
    def mostFrequent(self, nums: List[int], key: int) -> int:
        count = {}
        for i in range(len(nums) - 1):
            if nums[i] == key:
                if nums[i+1] not in count:
                    count[nums[i+1]] = 0
                count[nums[i+1]] += 1
        max_count = 0
        max_target = -1
        for target, c in count.items():
            if c > max_count:
                max_count = c
                max_target = target
        return max_target